fuel_gauge: sbs_gauge: fix negative currents

JordanYates opened the pull request:
Fix negative currents being read as extremely high currents due to unsigned variable use.

JordanYates committed:
fuel_gauge: sbs_gauge: fix negative currents

Fix negative currents being read as extremely high currents due to
unsigned variable use.

Signed-off-by: Jordan Yates <jordan@embeint.com>

aaronemassey reviewed (dismissed):
Well, that's surprising. That should be straight forward to add an emulated test if you'd like to add it.

JordanYates committed:
tests: fuel_gauge: test negative current

Validate that the SBS fuel gauge driver returns the expected values for
negative currents (discharging).

Signed-off-by: Jordan Yates <jordan@embeint.com>

JordanYates commented:
> Well, that's surprising. That should be straight forward to add an emulated test if you'd like to add it.

Test added, which fails without the fix.

aaronemassey reviewed (approved):
Awesome! Thank you for writing a test!
